### PR TITLE
Add helpers from the study platform

### PR DESCRIPTION
**Pull request by michaelheckmann**
# Add helpers from the study platform

## :recycle: Current situation & Problem
Some utils from the study platform could be useful in this repo.

## :gear: Release Notes
* Added the `Prettify` type in `misc.ts` to flatten complex intersection types, making them easier to read in tooling and type inspection.
* Added `joinPaths` function to `misc.ts`, which joins path segments while handling leading/trailing slashes, supports strings and numbers, and ignores empty/null/undefined segments.

## :white_check_mark: Testing
Added unit tests for `joinPaths` in `misc.test.ts`, covering various edge cases such as duplicate slashes, absolute URLs, query/hash preservation, and numeric segments. 

## :pencil: Code of Conduct & Contributing Guidelines
By creating and submitting this pull request, you agree to follow our [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md):
- [X] I agree to follow the [Code of Conduct](https://github.com/StanfordSpezi/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/StanfordSpezi/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: src/utils/misc/misc.ts [hunk 1]
```diff
@@ -70,6 +70,18 @@ export type PartialSome<T, K extends keyof T> = Omit<T, K> &
...
+ * Prettify makes complex object types easier to read by flattening intersections.
+ * Use this when you export or inspect inferred types. This only affects type presentation in tooling.
+ *
+ * @example
+ * type A = { a: string };
+ * type B = { b: number };
+ * type Mixed = A & B;
+ * type Readable = Prettify<Mixed>; // { a: string; b: number }
+ */
+export type Prettify<T> = { [K in keyof T]: T[K] } & {};
```
Comment (arkadiuszbachorski) [inline src/utils/misc/misc.ts:83]: Why is it simplified against previous implementation, without recurring object calls?

File: src/utils/misc/misc.ts [hunk 2]
```diff
@@ -188,3 +200,54 @@ export const formatBoolean = (value: boolean) => (value ? "true" : "false");
...
+ *
+ * @example
+ * joinPaths("https://example.com/", "/users/", "123"); // "https://example.com/users/123"
+ *
+ * @example
+ * joinPaths("/a", "b?x=1#top"); // "/a/b?x=1#top"
+ */
+export const joinPaths = (
+  ...segments: Array<string | number | null | undefined>
+): string => {
```
Comment (arkadiuszbachorski) [inline src/utils/misc/misc.ts:222]: Can we remove return type? It's easily inferable. 

```suggestion
) => {
```

File: src/utils/misc/misc.ts [hunk 3]
```diff
@@ -188,3 +200,54 @@ export const formatBoolean = (value: boolean) => (value ? "true" : "false");
...
+  const normalizedSegments = segments
+    .filter(
+      (segment): segment is string | number =>
+        segment != null && String(segment).length > 0,
+    )
```
Comment (arkadiuszbachorski) [inline src/utils/misc/misc.ts:227]: Can we replace `!= null` check with `!isNil`? My main motivation for this is removing loose comparison of `!=`, which can lead to misleading scenarios. 

```suggestion
    .filter(
      (segment): segment is string | number =>
        !isNil(segment) && String(segment).length > 0,
    )
```

File: src/utils/misc/misc.ts [hunk 4]
```diff
@@ -188,3 +200,54 @@ export const formatBoolean = (value: boolean) => (value ? "true" : "false");
...
+ * joinPaths("api", "users"); // "api/users"
+ *
+ * @example
+ * joinPaths("https://example.com/", "/users/", "123"); // "https://example.com/users/123"
+ *
+ * @example
+ * joinPaths("/a", "b?x=1#top"); // "/a/b?x=1#top"
+ */
+export const joinPaths = (
+  ...segments: Array<string | number | null | undefined>
```
Comment (arkadiuszbachorski) [inline src/utils/misc/misc.ts:221]:
```suggestion
  ...segments: Array<Nil<string | number>>
```

File: src/utils/misc/misc.ts [hunk 5]
```diff
@@ -188,3 +200,54 @@ export const formatBoolean = (value: boolean) => (value ? "true" : "false");
...
+): string => {
+  const normalizedSegments = segments
+    .filter(
+      (segment): segment is string | number =>
+        segment != null && String(segment).length > 0,
+    )
+    .map((segment) => String(segment))
```
Comment (arkadiuszbachorski) [inline src/utils/misc/misc.ts:228]:
```suggestion
    .map(String)
```

**Review by arkadiuszbachorski — CHANGES_REQUESTED** [review]
Thank you for adding that!

File: src/utils/misc/misc.ts (hunk 1 above)
Comment (arkadiuszbachorski) [reply src/utils/misc/misc.ts:83]: That's ok, I was just curious. We can address that in the future if needed.

**Review by arkadiuszbachorski — APPROVED** [review]

File: src/utils/misc/misc.ts [hunk 6]
```diff
@@ -188,3 +200,54 @@
...
+      const isLastSegment = index === arr.length - 1;
+
+      // Remove leading slashes from non-first segments to avoid "//" in the middle.
+      const withoutLeadingSlash =
+        isFirstSegment ? segment : segment.replace(/^\/+/, "");
+
+      // Remove trailing slashes from non-last segments to avoid duplicate separators.
+      const withoutTrailingSlash =
+        isLastSegment ? withoutLeadingSlash : (
+          withoutLeadingSlash.replace(/\/+$/, "")
```
[the remark this answers is not in the document]
Comment (michaelheckmann) [reply src/utils/misc/misc.ts:238]: Linear-time regex; no ReDoS
